gdkwayland/src/wayland_device.rs

```diff
@@ -31,28 +32,53 @@ impl WaylandDevice {
 
     #[doc(alias = "gdk_wayland_device_get_wl_seat")]
     #[doc(alias = "get_wl_seat")]
-    pub fn wl_seat(&self) -> WlSeat {
+    pub fn wl_seat(&self) -> Option<WlSeat> {
+        let display = self.display().downcast::<crate::WaylandDisplay>().unwrap();
```

Comment:
You could `unsafe_cast()` here to reduce the amount of generated code

Reply:
(also elsewhere)

Reply:
Also you only need the display if the C function below returns non-`NULL` so you can do it afterwards